src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
+        if (app_data.keyExists(json::android)) {
+          os_type = json::android;
+          if (!app_data[os_type].keyExists(json::packageName)) {
+            LOG4CXX_WARN(logger_,
+                         error_prefix_
+                             << "Can't find package name in json file.");
+            result = false;
+          } else {
+            if (app_data[json::android][json::packageName].asString().length() >
+                max_package_name_length_) {
+              LOG4CXX_WARN(logger_,
+                           error_prefix_
+                               << "Package name length ["
+                               << app_data[json::android][json::packageName]
+                                      .asString()
+                                      .length() << "] exceeds max length ["
+                               << max_package_name_length_ << "]in json file.");
+              result = false;
+            }
+          }
+        }
+      }
+
+      if (os_type.empty()) {
+        LOG4CXX_WARN(logger_,
+                     error_prefix_
+                         << "Can't find mobile OS type in json file.");
+        result = false;
+        continue;
+      }
+
+      // Languages verification
+      if (!app_data[os_type].keyExists(json::languages)) {
+        LOG4CXX_WARN(logger_, error_prefix_ << "\"languages\" not exists");
+        result = false;
+        continue;
+      }
+      if (!ValidateLanguages(app_data[os_type][json::languages]))
+        result = false;
+    }
+    return result;
+  }
+
+  bool ValidateAppIdAndAppName(
+      const smart_objects::SmartObject& app_data) const {
+    // Verify appid
+    if (!app_data.keyExists(json::appId)) {
+      LOG4CXX_WARN(logger_, error_prefix_ << "Can't find app ID in json file.");
+      return false;
+    }
+    bool result = true;
+    // Verify appid lenght
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > max_app_id_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[" << max_app_id_length_
+                       << "]");
+      result = false;
+    }
+
+    // Verify that appid is unique
+    if (app_ids_map_.find(app_id) != app_ids_map_.end()) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "An Object ID is not unigue [" << app_id
+                                 << "]");
+      result = false;
+    } else {
+      app_ids_map_[app_id] = 1;
+    }
+
+    // Verify that app is not registered yet
+    ApplicationSharedPtr registered_app =
+        ApplicationManagerImpl::instance()->application_by_policy_id(app_id);
+    if (registered_app) {
+      LOG4CXX_DEBUG(logger_,
+                    error_prefix_ << "Application with the same id: " << app_id
+                                  << " is registered already.");
+      result = false;
```

Comment:
@okoreniak, am I right that this param is optional, that is why we don't return and log just debug?